src/rsa.rs

```diff
@@ -116,6 +116,25 @@ fn verify(padding: &RSAPadding, msg: untrusted::Input,
 }
 
 
+pub mod primitive {
+    use untrusted;
+    /// Lower-level verification primitive for RSA signature verification.
+    ///
+    /// Takes an `RSAPadding` static to determine the padding algorithm used,
+    /// a message, a signature (which should match the length of public
+    /// `n`), the parts of the public key (in big-endian byte order). The key
+    /// length must be between 2048 and 8192 bits.
+    pub fn verify(padding: &super::RSAPadding, msg: untrusted::Input,
+                  signature: untrusted::Input,
+                  (n, e): (untrusted::Input, untrusted::Input))
+                  -> Result<(), ()> {
+        let n = n.as_slice_less_safe();
+        let e = e.as_slice_less_safe();
```

Comment:
I think instead we should change `parse_public_key` so that it returns `untrusted::Input`s and then have `super::verify` take `(n, e): (untrusted::Input, untrusted::Input)`. This removes the duplication of the `as_slice_less_safe` and fits the pattern I'm trying to promote wherein we avoid using slices for untrusted inputs when possible.